libs/vm-modules/src/math/tensor.cpp

```diff
@@ -57,28 +63,85 @@ Ptr<VMTensor> VMTensor::Constructor(VM *vm, TypeId type_id, Ptr<Array<SizeType>>
 
 void VMTensor::Bind(Module &module)
 {
+
+  ///////////////////////////////
+  // Prepare charge estimators //
+  ///////////////////////////////
+
+  // scalar estimation for cheap operations
+  ChargeAmount low_charge{vm::CHARGE_UNIT};
+
+  // object state-dependent estimators
+  auto charge_func_of_tensor_size = [](Ptr<VMTensor> const &this_,
```

Comment:
move this to a free function in an anonymous namespace in this file - then your estimators won't have to drag a copy around and you won't have to capture it everywhere.